x/merkledb/history.go

```diff
@@ -183,23 +179,25 @@ func (th *trieHistory) getValueChanges(startRoot, endRoot ids.ID, start []byte,
 					bytes.Equal(existing.before.Value(), existing.after.Value()) {
 					// The change to this key is a no-op, so remove it from [combinedChanges].
 					delete(combinedChanges.values, key)
-					sortedKeys.Delete(key)
+					changedKeys.Remove(key)
 				}
 			} else {
 				combinedChanges.values[key] = &change[maybe.Maybe[[]byte]]{
 					before: valueChange.before,
 					after:  valueChange.after,
 				}
-				sortedKeys.ReplaceOrInsert(key)
+				changedKeys.Add(key)
 			}
 		}
 	}
 
 	// Keep only the smallest [maxLength] items in [combinedChanges.values].
-	for sortedKeys.Len() > maxLength {
-		if greatestKey, found := sortedKeys.DeleteMax(); found {
-			delete(combinedChanges.values, greatestKey)
-		}
+	sortedChangedKeys := changedKeys.List()
+	utils.Sort(sortedChangedKeys)
```

Comment:
can avoid the sort if the total change length < max change Length

Reply:
Good point, done